engine/orchestration/src/com/cloud/vm/VirtualMachineManagerImpl.java

```diff
@@ -599,6 +607,60 @@ protected void advanceExpunge(VMInstanceVO vm) throws ResourceUnavailableExcepti
 
     }
 
+    private List<Map<String, String>> getTargets(Long hostId, long vmId) {
+        List<Map<String, String>> targets = new ArrayList<>();
+
+        HostVO hostVO = _hostDao.findById(hostId);
+
+        if (hostVO != null && hostVO.getHypervisorType() == HypervisorType.VMware) {
+            List<VolumeVO> volumes = _volsDao.findByInstance(vmId);
+
+            if (volumes != null) {
```

Comment:
what if you invert this one here?
`if (CollectionUtils.isEmpty(volumes))`, then you return targets.
This would allow to remove the other for and IF statement from one conditional.

Rule of thumb: the less nested things, the better to read.

Reply:
Done. My initial approach here was due to the fact that some developers/reviewers don't like multiple return points in a method. Personally, I'm fine with what you suggest here and have updated the code.

Reply:
Got it. I have seen reviewers also saying that. I personally disagree.... specially in method where we might have few conditionals, and by doing this check and return, we can avoid nesting them.

Thanks for your work!